client/src/controllers/TabsController.ts

```diff
@@ -0,0 +1,380 @@
+ * All tab buttons need the role="tab" attr and an href with the tab content ID with the target 'label'.
+ * Tab contents need to have the role="tabpanel" attribute and and ID attribute that matches the href of the tab link with the target 'panel'.
+ * Tab buttons should also be wrapped in an element with the role="tablist" attribute.
+ * Use the target 'trigger' on an Anchor link and set the href to the #ID of the tab you would like to trigger.
+ *
+ * @example
+ * ```html
+ * <div data-controller="w-tabs" data-action="popstate@window->w-tabs#loadHistory" data-w-tabs-selected-class="animate-in">
+ *   <div role="tablist" data-action="click->w-tabs#handleTabChange:prevent keydown.right->w-tabs#selectNext keydown.left->w-tabs#selectPrevious keydown.home->w-tabs#selectFirst keydown.end->w-tabs#selectLast">
+ *     <a id="tab-label-tab-1" href="#tab-tab-1" role="tab" data-w-tabs-target="label">Tab 1</a>
+ *     <a id="tab-label-tab-2" href="#tab-tab-2" role="tab" data-w-tabs-target="label">Tab 2</a>
+ *  </div>
+ *    <div class="tab-content tab-content--comments-enabled">
+ *     <section id="tab-tab-1" role="tabpanel" aria-labelledby="tab-label-tab-1" data-w-tabs-target="panel">
+ *       Tab 1 content
+ *     </section>
+ *     <section id="tab-tab-2" role="tabpanel" aria-labelledby="tab-label-tab-2" data-w-tabs-target="panel">
+ *       Tab 2 content
+ *     </section>
+ *   </div>
+ * </div>
+ * ```
+ */
+
+export class TabsController extends Controller<HTMLDivElement> {
+  static targets = ['label', 'panel', 'trigger'];
+
+  static classes = ['selected'];
+
+  static values = {
+    transition: { default: 150, type: Number },
+    selected: { default: '', type: String },
+    syncURLHash: { default: false, type: Boolean },
+    animate: { default: true, type: Boolean },
+  };
+
+  /** ID of the currently selected tab. */
+  declare selectedValue: string;
+  /** If true, animation will run when a new tab is selected. */
+  declare readonly animateValue: boolean;
+  /** Tab elements, with role='tab', allowing switching between tabs. */
+  declare readonly labelTargets: HTMLAnchorElement[];
+  /** Tab content panels, with role='tabpanel', showing the content for each tab. */
+  declare readonly panelTargets: HTMLElement[];
+  /** Other elements within the controller's scope that may trigger a specific tab. */
+  declare readonly triggerTargets: HTMLAnchorElement[];
+  /** Classes to set on the tab panel content when selected. */
+  declare readonly selectedClasses: string[];
+  /** If true, the selected tab will sync with the URL hash. */
+  declare readonly syncURLHashValue: boolean;
+  /** The time in milliseconds for the tab content to transition in and out. */
+  declare readonly transitionValue: number;
+
+  connect() {
+    this.validate();
+
+    debounce(() => {
+      window.scrollTo({ top: 0, behavior: 'smooth' });
+    }, this.transitionValue * 2);
```

Comment:
This code will not do anything as is, you either need to call the result of debounce...

```suggestion
    debounce(() => {
      window.scrollTo({ top: 0, behavior: 'smooth' });
    }, this.transitionValue * 2)();
```

However, this is probably not what we want, this will mean that each new tabs element that appears on screen will cause the window to scroll to the top when it loads.

For now let's go with and maybe @thibaudcolas  can shed some light on this and whether this makes sense now that tabs are used everywhere, not just for the page editor.